loki/backend/cgen.py

```diff
@@ -364,6 +365,35 @@ def visit_TypeDef(self, o, **kwargs):
         self.depth -= 1
         return self.join_lines(header, decls, footer)
 
+    def visit_MultiConditional(self, o, **kwargs):
+        """
+        Format as
+          switch case (<expr>) {
+          case <value>:
+            ...body...
+          [case <value>:]
+            [...body...]
+          [default:]
+            [...body...]
+          }
+        """
+        header = self.format_line('switch (', self.visit(o.expr, **kwargs), ') {')
+        cases = []
+        end_cases = []
+        for value in o.values:
+            case = self.visit_all(as_tuple(value), **kwargs)
```

Comment:
Note that you'll likely run into issues when the cases are ranges (which is allowed in Fortran: https://j3-fortran.org/doc/year/10/10-007.pdf#page=200). These are likely not directly transferable to C, but maybe it would be useful to throw an exception if a `RangeIndex` (I presume that's what we would use to represent them here) is encountered?

Reply:
Lovely ... thank you for pointing this out to me